### PR TITLE
feat(frontend): util for tokens total USD balance calculation

### PR DESCRIPTION
**Pull request by DenysKarmazynDFINITY**
# Motivation

Step 1 of updating the chain selector with aggregate USD balance values: re-use util for calculation tokens total USD balance. In the next steps, it will be used to calculate this value for different networks (as shown on the screenshot).

Screenshot 2024-09-03 at 13 39 05

### CODE REVIEW

File: src/frontend/src/lib/components/exchange/ExchangeBalance.svelte [hunk 1]
```diff
@@ -2,12 +2,10 @@
...
 	import { exchangeInitialized } from '$lib/derived/exchange.derived';
 	import { combinedDerivedSortedNetworkTokensUi } from '$lib/derived/network-tokens.derived';
+	import { getTokensTotalUsdBalance } from '$lib/utils/tokens.utils';
 
 	let totalUsd: number;
-	$: totalUsd = $combinedDerivedSortedNetworkTokensUi.reduce(
-		(acc, token) => acc + (token.usdBalance ?? 0),
-		0
-	);
+	$: totalUsd = getTokensTotalUsdBalance($combinedDerivedSortedNetworkTokensUi);
```
Comment (peterpeterparker) [inline src/frontend/src/lib/components/exchange/ExchangeBalance.svelte:8]: it's not really a getter per sé, maybe we can rename it to something like `sumTotalUsdBalance` or better idea?

File: src/frontend/src/lib/utils/tokens.utils.ts [hunk 2]
```diff
@@ -80,3 +80,11 @@ export const pinTokensWithBalanceAtTop = ($tokens: TokenUi[]): TokenUi[] => {
 		...nonPositiveBalances
 	];
 };
+
+/**
+ * Calculates total USD balance of the provided tokens list.
+ *
+ * @param $tokens - The list of tokens for total USD balance calculation.
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/tokens.utils.ts:87]: If you add the documentation then maybe also document the return type?

File: src/frontend/src/lib/utils/tokens.utils.ts [hunk 3]
```diff
@@ -80,3 +80,11 @@ export const pinTokensWithBalanceAtTop = ($tokens: TokenUi[]): TokenUi[] => {
...
+ */
+export const getTokensTotalUsdBalance = ($tokens: TokenUi[]): number =>
```
Comment (peterpeterparker) [inline src/frontend/src/lib/utils/tokens.utils.ts:89]: I think @AntonioVentilii-DFINITY uses the `$` pattern to express the fact that the variable is a store. In this case it isn't one so I would remove this prefix but, Antonio can confirm?

File: src/frontend/src/tests/lib/utils/tokens.utils.spec.ts [hunk 4]
```diff
@@ -156,3 +160,16 @@ describe('pinTokensWithBalanceAtTop', () => {
 		]);
 	});
 });
+
+describe('getTokensTotalUsdBalance', () => {
+	it('should correctly calculate tokens USD total balance', () => {
```
Comment (peterpeterparker) [inline src/frontend/src/tests/lib/utils/tokens.utils.spec.ts:165]: Maybe add a test if the previded list is empty?
And another if a token has no usdBalance?

**Review by peterpeterparker — COMMENTED** [review]
Jira is not public, so not really useful in the content of open source. Could you maybe edit the PR description to remove this reference?

File: src/frontend/src/lib/components/exchange/ExchangeBalance.svelte (hunk 1 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/components/exchange/ExchangeBalance.svelte]: Agreed. 
I renamed it to `sumTokensUsdBalance` to highlight that a list of tokens is expected as param and also removed the `Total` part since `sum` already means that there will be a total. WDYT?

File: src/frontend/src/lib/utils/tokens.utils.ts (hunk 2 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts:87]: Done.

File: src/frontend/src/tests/lib/utils/tokens.utils.spec.ts (hunk 4 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/tests/lib/utils/tokens.utils.spec.ts]: Sounds good. Added 2 additional test cases (empty list + list with tokens w/o `usdBalance`).

File: src/frontend/src/lib/utils/tokens.utils.ts (hunk 3 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts]: True, this helper can possible called for any list of tokens, not necessarily a store value from `network-tokens.derived`. @AntonioVentilii-DFINITY fine with you if I remove `$` here?

File: src/frontend/src/lib/components/exchange/ExchangeBalance.svelte (hunk 1 above)
Comment (peterpeterparker) [reply src/frontend/src/lib/components/exchange/ExchangeBalance.svelte]: Coolio, thanks.

File: src/frontend/src/lib/utils/tokens.utils.ts (hunk 3 above)
Comment (AntonioVentilii-DFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts]: If the scope is not purely for stores or drives, I agree, you can remove the dollar sign

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks

Already approving, that way you can move on once the `$` removed as discussed.

File: src/frontend/src/lib/utils/tokens.utils.ts (hunk 3 above)
Comment (DenysKarmazynDFINITY) [reply src/frontend/src/lib/utils/tokens.utils.ts]: Renamed 👍
